### PR TITLE
fix(bridge-ui): transaction and pendingTransaction refactor

### PR DESCRIPTION
**Pull request by shadab-taiko**

### CODE REVIEW

File: packages/bridge-ui/src/App.svelte [hunk 1]
```diff
@@ -174,20 +174,20 @@
   });
 
   pendingTransactions.subscribe((store) => {
-    store.forEach(async (tx) => {
-      await $signer.provider.waitForTransaction(tx.hash, 1);
+    (async () => {
+      const confirmedPendingTxIndex = await Promise.race(
```
Comment (shadab-taiko) [inline packages/bridge-ui/src/App.svelte:178]: This way, we wouldn't be having stale `Promise`s. Earlier stale `Promise`s were resolving and each was triggering a toast which resulted in more than one toast being shown for each pending transaction.

File: packages/bridge-ui/src/relayer-api/service.ts [hunk 2]
```diff
@@ -116,7 +110,9 @@ class RelayerAPIService implements RelayerAPI {
         );
 
         const event = events.find(
-          (e) => e.args.message.owner.toLowerCase() === address.toLowerCase(),
+          (e) =>
+            e.args.message.owner.toLowerCase() === address.toLowerCase() &&
+            e.args.message.depositValue.eq(tx.message.depositValue),
```
Comment (shadab-taiko) [inline packages/bridge-ui/src/relayer-api/service.ts:115]: For transactions included within the same block we were picking the first event out of many which resulted in wrong data being shown in the transaction list. This fixes it.

Comment (jscriptcoder) [reply packages/bridge-ui/src/relayer-api/service.ts:115]: Maybe we could add that as a comment?

File: packages/bridge-ui/src/vite-env.d.ts [hunk 3]
```diff
@@ -1,2 +1,32 @@
...
+  readonly VITE_MAINNET_HEADER_SYNC_ADDRESS: string;
+  readonly VITE_TAIKO_HEADER_SYNC_ADDRESS: string;
+  readonly VITE_MAINNET_BRIDGE_ADDRESS: string;
+  readonly VITE_TAIKO_BRIDGE_ADDRESS: string;
+  readonly VITE_MAINNET_SIGNAL_SERVICE_ADDRESS: string;
+  readonly VITE_TAIKO_SIGNAL_SERVICE_ADDRESS: string;
+  readonly VITE_TEST_ERC20: string;
+}
+
+interface ImportMeta {
```
Comment (jscriptcoder) [inline packages/bridge-ui/src/vite-env.d.ts:30]: Nice one

**Review by jscriptcoder — DISMISSED** [review]

**Review by cyberhorsey — DISMISSED** [review]

File: packages/bridge-ui/src/relayer-api/service.ts (hunk 2 above)
Comment (shadab-taiko) [reply packages/bridge-ui/src/relayer-api/service.ts]: I just realized that even the depositValue could be the exact same. So we need to also compare `msgHash`.

**Review by jscriptcoder — APPROVED** [review]

**Review by davidtaikocha — APPROVED** [review]

**Pull request merged**